{"ast":null,"code":"'use strict';\n\n// Try getting the real promise object from the context, if available. Someone\n// could have overridden it in a test.\nvar _createClass = require(\"C:\\\\Users\\\\legoj\\\\OneDrive\\\\Dokument\\\\Uni\\\\VT 2023\\\\Webbteknik\\\\hemsidan\\\\reactp\\\\myprojectapp\\\\node_modules\\\\babel-preset-react-app\\\\node_modules\\\\@babel\\\\runtime/helpers/createClass\");\nvar _classCallCheck = require(\"C:\\\\Users\\\\legoj\\\\OneDrive\\\\Dokument\\\\Uni\\\\VT 2023\\\\Webbteknik\\\\hemsidan\\\\reactp\\\\myprojectapp\\\\node_modules\\\\babel-preset-react-app\\\\node_modules\\\\@babel\\\\runtime/helpers/classCallCheck\");\nvar _possibleConstructorReturn = require(\"C:\\\\Users\\\\legoj\\\\OneDrive\\\\Dokument\\\\Uni\\\\VT 2023\\\\Webbteknik\\\\hemsidan\\\\reactp\\\\myprojectapp\\\\node_modules\\\\babel-preset-react-app\\\\node_modules\\\\@babel\\\\runtime/helpers/possibleConstructorReturn\");\nvar _getPrototypeOf = require(\"C:\\\\Users\\\\legoj\\\\OneDrive\\\\Dokument\\\\Uni\\\\VT 2023\\\\Webbteknik\\\\hemsidan\\\\reactp\\\\myprojectapp\\\\node_modules\\\\babel-preset-react-app\\\\node_modules\\\\@babel\\\\runtime/helpers/getPrototypeOf\");\nvar _inherits = require(\"C:\\\\Users\\\\legoj\\\\OneDrive\\\\Dokument\\\\Uni\\\\VT 2023\\\\Webbteknik\\\\hemsidan\\\\reactp\\\\myprojectapp\\\\node_modules\\\\babel-preset-react-app\\\\node_modules\\\\@babel\\\\runtime/helpers/inherits\");\nvar _wrapNativeSuper = require(\"C:\\\\Users\\\\legoj\\\\OneDrive\\\\Dokument\\\\Uni\\\\VT 2023\\\\Webbteknik\\\\hemsidan\\\\reactp\\\\myprojectapp\\\\node_modules\\\\babel-preset-react-app\\\\node_modules\\\\@babel\\\\runtime/helpers/wrapNativeSuper\");\nvar Promise = global[Symbol.for('jest-native-promise')] || global.Promise;\nvar CancelError = /*#__PURE__*/function (_Error) {\n  _inherits(CancelError, _Error);\n  function CancelError() {\n    var _this;\n    _classCallCheck(this, CancelError);\n    _this = _possibleConstructorReturn(this, _getPrototypeOf(CancelError).call(this, 'Promise was canceled'));\n    _this.name = 'CancelError';\n    return _this;\n  }\n  return CancelError;\n}(_wrapNativeSuper(Error));\nvar PCancelable = /*#__PURE__*/function () {\n  _createClass(PCancelable, null, [{\n    key: \"fn\",\n    value: function fn(_fn) {\n      return function () {\n        var args = [].slice.apply(arguments);\n        return new PCancelable(function (onCancel, resolve, reject) {\n          args.unshift(onCancel);\n          _fn.apply(null, args).then(resolve, reject);\n        });\n      };\n    }\n  }]);\n  function PCancelable(executor) {\n    var _this2 = this;\n    _classCallCheck(this, PCancelable);\n    this._pending = true;\n    this._canceled = false;\n    this._promise = new Promise(function (resolve, reject) {\n      _this2._reject = reject;\n      return executor(function (fn) {\n        _this2._cancel = fn;\n      }, function (val) {\n        _this2._pending = false;\n        resolve(val);\n      }, function (err) {\n        _this2._pending = false;\n        reject(err);\n      });\n    });\n  }\n  _createClass(PCancelable, [{\n    key: \"then\",\n    value: function then() {\n      return this._promise.then.apply(this._promise, arguments);\n    }\n  }, {\n    key: \"catch\",\n    value: function _catch() {\n      return this._promise.catch.apply(this._promise, arguments);\n    }\n  }, {\n    key: \"cancel\",\n    value: function cancel() {\n      if (!this._pending || this._canceled) {\n        return;\n      }\n      if (typeof this._cancel === 'function') {\n        try {\n          this._cancel();\n        } catch (err) {\n          this._reject(err);\n        }\n      }\n      this._canceled = true;\n      this._reject(new CancelError());\n    }\n  }, {\n    key: \"canceled\",\n    get: function get() {\n      return this._canceled;\n    }\n  }]);\n  return PCancelable;\n}();\nObject.setPrototypeOf(PCancelable.prototype, Promise.prototype);\nmodule.exports = PCancelable;\nmodule.exports.CancelError = CancelError;","map":null,"metadata":{},"sourceType":"script"}